common/execution_context.go

```diff
@@ -206,7 +206,9 @@ func (e *ExecutionContext) eval(
 	)
 	if remoteObject, exceptionDetails, err = action.Do(cdp.WithExecutor(apiCtx, e.session)); err != nil {
 		var cdpe *cdproto.Error
-		if errors.As(err, &cdpe) && cdpe.Code == -32000 {
+		if errors.As(err, &cdpe) && cdpe.Message == "Given expression does not evaluate to a function" {
+			err = errors.New("given expression does not evaluate to a function")
+		} else if errors.As(err, &cdpe) && cdpe.Code == -32000 {
```

Comment:
Is there a constant in the cdp package for this (-32000)?

Reply:
Unfortunately not, it seems to originate from chromium [here](https://chromium.googlesource.com/chromium/src/+/665cdea2f5f29e166ad455b5021154456944fcbc/headless/lib/browser/headless_devtools_manager_delegate.cc#46) and [here](https://chromium.googlesource.com/chromium/src/third_party/WebKit/Source/platform/v8_inspector/+/d94ef126e8fd904cedea7bca029216e42fe9eb6a/V8RuntimeAgentImpl.cpp#377) for the error message.


Take a look at this [file](https://chromium.googlesource.com/chromium/src/+/665cdea2f5f29e166ad455b5021154456944fcbc/headless/lib/browser/headless_devtools_manager_delegate.cc). After searching for `kErrorServerError`, which is the enum name of error code `-32000`, it looks like we're handling that error incorrectly. Maybe we should be returning the error message instead of `execution context changed; most likely because of a navigation`, WDYT?

Reply:
Nice digging into the Chromium source code. Thanks for the links 👍 

I would, then, declare a constant to represent this in a `devToolsServerErrorCode` constant for clarity. And, for this part: `unexpected non server error: %`; what about the following: `Unexpected DevTools server error: %v`.  WDYT?

One last thing. We can return the first case without using an `else`.

Reply:
Resolved in 9277d6013373403399da56efea13461dde4ef39f